src/core/lib/iomgr/port.h

```diff
@@ -37,6 +37,15 @@
 #define GRPC_HAVE_IP_PKTINFO 1
 #define GRPC_HAVE_MSG_NOSIGNAL 1
 #define GRPC_HAVE_UNIX_SOCKET 1
+#ifdef LINUX_VERSION_CODE
```

Comment:
nit: I think short-circuit macro evaluation would work here, since we only care about the inner-most scope.

```
#if defined(LINUX_VERSION_CODE) && LINUX_VERSION_CODE >= KERNEL_VERSION(3, 9, 0) && \
    defined(__GLIBC_PREREQ) && __GLIBC_PREREQ(2, 18)
#define GRPC_HAVE_VSOCK 1
#endif

Reply:
Updated, please review.